### ISSUE TITLE
SQL Generation for Custom Changes no longer working in 4.22.0

### ISSUE THREAD

**Issue by tarioch** [liquibase/liquibase#4369]
### Search first

- [X] I searched and no similar issues were found

### Description

Since #4054 custom sql changes do no longer generate code if they are run with updateSQL mode, applying the changes with update works fine.

### Steps To Reproduce

I've created a reproduction case here:

https://github.com/tarioch/liquibase-issue

When running ```mvn liquibase:updateSQL``` it doesn't generate the custom sql change, with a version before #4054 it works fine and generates it.

### Expected/Desired Behavior

Custom SQL Change is also generated in updateSQL mode

### Liquibase Version

4.22.0

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

**Comment by tati-qalified** [comment]
Hi @tarioch , how are you?
[Liquibase version 4.23.0](https://github.com/liquibase/liquibase/releases/tag/v4.23.0) just came out, so I ran your code with the updated version, and I was able to get it to work.
Just in case, I created my own custom changeset to run alongside yours, and the code for both was created:

<img width="769" alt="Capture" src="https://github.com/liquibase/liquibase/assets/135264316/10615d79-e77d-4a57-bb2f-e1c4c572e5ed">

As you can see, the code for two custom changesets is generated.

In conclusion, try updating to the latest Liquibase version, and it should work for you.
I will be closing this ticket, though be sure to reopen it if the issue persists.

**Comment by tarioch** [comment]
Hi @tati-qalified 

looking at your screenshot the issue is still present. The sql for the custom change is missing. The insert into the changelog was always there, the issue is that the actual custom sql change is missing in generation mode.

**Comment by tarioch** [comment]
So on 4.20.0 the produced sql is

```sql
-- *********************************************************************
-- Update Database Script
-- *********************************************************************
-- Change Log: changelog.xml
-- Ran at: 6/29/23, 12:57 PM
-- Against: null@jdbc:sqlite:example.db
-- Liquibase version: 4.20.0
-- *********************************************************************

-- Create Database Lock Table
CREATE TABLE DATABASECHANGELOGLOCK (ID INTEGER NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TEXT, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID));

-- Initialize Database Lock Table
DELETE FROM DATABASECHANGELOGLOCK;

INSERT INTO DATABASECHANGELOGLOCK (ID, LOCKED) VALUES (1, 0);

-- Lock Database
UPDATE DATABASECHANGELOGLOCK SET LOCKED = 1, LOCKEDBY = 'vwmazapxvd-1 (10.7.175.137)', LOCKGRANTED = CURRENT_TIMESTAMP WHERE ID = 1 AND LOCKED = 0;

-- Create Database Change Log Table
CREATE TABLE DATABASECHANGELOG (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TEXT NOT NULL, ORDEREXECUTED INTEGER NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10));

-- Changeset changelog.xml::1::Liquibase
-- WARNING The following SQL may change each run and therefore is possibly incorrect and/or invalid:
exec myDummy();

INSERT INTO DATABASECHANGELOG (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('1', 'Liquibase', 'changelog.xml', CURRENT_TIMESTAMP, 1, '8:e7c4a15d2f0309bb7124d88c1e74eb80', 'customChange', '', 'EXECUTED', NULL, NULL, '4.20.0', '8036244405');

-- Release Database Lock
UPDATE DATABASECHANGELOGLOCK SET LOCKED = 0, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1;
```

using 4.23.0, it looks like this

```sql
-- Create Database Lock Table
CREATE TABLE DATABASECHANGELOGLOCK (ID INTEGER NOT NULL, LOCKED BOOLEAN NOT NULL, LOCKGRANTED TEXT, LOCKEDBY VARCHAR(255), CONSTRAINT PK_DATABASECHANGELOGLOCK PRIMARY KEY (ID));

-- Initialize Database Lock Table
DELETE FROM DATABASECHANGELOGLOCK;

INSERT INTO DATABASECHANGELOGLOCK (ID, LOCKED) VALUES (1, 0);

-- Lock Database
UPDATE DATABASECHANGELOGLOCK SET LOCKED = 1, LOCKEDBY = 'vwmazapxvd-1 (10.7.175.137)', LOCKGRANTED = CURRENT_TIMESTAMP WHERE ID = 1 AND LOCKED = 0;

-- Create Database Change Log Table
CREATE TABLE DATABASECHANGELOG (ID VARCHAR(255) NOT NULL, AUTHOR VARCHAR(255) NOT NULL, FILENAME VARCHAR(255) NOT NULL, DATEEXECUTED TEXT NOT NULL, ORDEREXECUTED INTEGER NOT NULL, EXECTYPE VARCHAR(10) NOT NULL, MD5SUM VARCHAR(35), DESCRIPTION VARCHAR(255), COMMENTS VARCHAR(255), TAG VARCHAR(255), LIQUIBASE VARCHAR(20), CONTEXTS VARCHAR(255), LABELS VARCHAR(255), DEPLOYMENT_ID VARCHAR(10));

-- *********************************************************************
-- Update Database Script
-- *********************************************************************
-- Change Log: changelog.xml
-- Ran at: 6/29/23, 12:59 PM
-- Against: null@jdbc:sqlite:example.db
-- Liquibase version: 4.23.0
-- *********************************************************************

-- Changeset changelog.xml::1::Liquibase
-- WARNING The following SQL may change each run and therefore is possibly incorrect and/or invalid:
INSERT INTO DATABASECHANGELOG (ID, AUTHOR, FILENAME, DATEEXECUTED, ORDEREXECUTED, MD5SUM, DESCRIPTION, COMMENTS, EXECTYPE, CONTEXTS, LABELS, LIQUIBASE, DEPLOYMENT_ID) VALUES ('1', 'Liquibase', 'changelog.xml', CURRENT_TIMESTAMP, 1, '9:cac71f1adc2b0955308a7cd9786eca3f', 'customChange', '', 'EXECUTED', NULL, NULL, '4.23.0', '8036372463');

-- Release Database Lock
UPDATE DATABASECHANGELOGLOCK SET LOCKED = 0, LOCKEDBY = NULL, LOCKGRANTED = NULL WHERE ID = 1;
```

As you can see the actual custom sql change is missing

```sql
-- Changeset changelog.xml::1::Liquibase
-- WARNING The following SQL may change each run and therefore is possibly incorrect and/or invalid:
exec myDummy();
```

**Comment by kevin-atx** [comment]
@filipelautert - could you check and see if this is a new error in the Java API?

**Comment by filipelautert** [comment]
PR https://github.com/liquibase/liquibase/pull/4424 that was merged on 4.25.0 was aimed to fix this. Could you confirm it works @tarioch ?

**Comment by tarioch** [comment]
Yes, I can confirm that now it's fixed, thank you very much
